test/e2e/metrics_runtime_node_namespace_test.go

```diff
@@ -0,0 +1,213 @@
+					))
+				}, periodic.TelemetryEventuallyTimeout, periodic.TelemetryInterval).Should(Succeed())
+			})
+		})
+	})
+
+	Context("When metric pipelines with node metrics enabled and an exclude namespace selector exist", Ordered, func() {
+		var (
+			excludeNS = suite.IDWithSuffix("exclude-ns")
+
+			backendOnlyNodeMetricsEnabledName  = suite.IDWithSuffix("node-metrics")
+			pipelineOnlyNodeMetricsEnabledName = suite.IDWithSuffix("node-metrics")
+			backendOnlyNodeMetricsEnabledURL   string
+		)
+
+		makeResources := func() []client.Object {
+			var objs []client.Object
+			objs = append(objs, kitk8s.NewNamespace(excludeNS).K8sObject())
+
+			backendOnlyNodeMetricsEnabled := backend.New(excludeNS, backend.SignalTypeMetrics, backend.WithName(backendOnlyNodeMetricsEnabledName))
+			objs = append(objs, backendOnlyNodeMetricsEnabled.K8sObjects()...)
+			backendOnlyNodeMetricsEnabledURL = backendOnlyNodeMetricsEnabled.ExportURL(proxyClient)
+
+			pipelineOnlyNodeMetricsEnabledWithIncludeNS := testutils.NewMetricPipelineBuilder().
+				WithName(pipelineOnlyNodeMetricsEnabledName).
+				WithRuntimeInput(true,
+					testutils.ExcludeNamespaces(excludeNS),
+				).
+				WithRuntimeInputNodeMetrics(true).
+				WithRuntimeInputPodMetrics(false).
+				WithRuntimeInputContainerMetrics(false).
+				WithRuntimeInputVolumeMetrics(false).
+				WithOTLPOutput(testutils.OTLPEndpoint(backendOnlyNodeMetricsEnabled.Endpoint())).
+				Build()
+			objs = append(objs, &pipelineOnlyNodeMetricsEnabledWithIncludeNS)
+
+			metricProducer := prommetricgen.New(excludeNS)
+
+			objs = append(objs, []client.Object{
+				metricProducer.Pod().WithPrometheusAnnotations(prommetricgen.SchemeHTTP).K8sObject(),
+				metricProducer.Service().WithPrometheusAnnotations(prommetricgen.SchemeHTTP).K8sObject(),
+			}...)
+
+			return objs
+		}
+
+		BeforeAll(func() {
+			k8sObjects := makeResources()
+
+			DeferCleanup(func() {
+				Expect(kitk8s.DeleteObjects(ctx, k8sClient, k8sObjects...)).Should(Succeed())
+			})
+			Expect(kitk8s.CreateObjects(ctx, k8sClient, k8sObjects...)).Should(Succeed())
+		})
+
+		It("Should have healthy pipelines", func() {
+			assert.MetricPipelineHealthy(ctx, k8sClient, pipelineOnlyNodeMetricsEnabledName)
+		})
+
+		It("Ensures the metric gateway deployment is ready", func() {
+			assert.DeploymentReady(ctx, k8sClient, kitkyma.MetricGatewayName)
+		})
+
+		It("Ensures the metric agent daemonset is ready", func() {
+			assert.DaemonSetReady(ctx, k8sClient, kitkyma.MetricAgentName)
+		})
+
+		It("Should have metrics backends running", func() {
+			assert.DeploymentReady(ctx, k8sClient, types.NamespacedName{Name: backendOnlyNodeMetricsEnabledName, Namespace: excludeNS})
+			assert.ServiceReady(ctx, k8sClient, types.NamespacedName{Name: backendOnlyNodeMetricsEnabledName, Namespace: excludeNS})
+
+		})
+
+		Context("Runtime node metrics", func() {
```

Comment:
is there a reason for having the same context duplicated, i.e. "Runtime node metrics"?

Reply:
refactored the code. Playing around with table tests. Let me know what you think about that approach.

Reply:
Yeah, I like it. Let's keep it like this